File: VB/ConfirmTabClose/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraTab
Imports DevExpress.XtraTab.ViewInfo

Namespace ConfirmTabClose
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub xtraTabControl1_CloseButtonClick(ByVal sender As Object, ByVal e As EventArgs) Handles xtraTabControl1.CloseButtonClick
			Dim tabControl As XtraTabControl = CType(sender, XtraTabControl)
			Dim arg As ClosePageButtonEventArgs = CType(e, ClosePageButtonEventArgs)
			If MessageBox.Show(String.Format("Close {0}?", arg.Page.Text), "Confirmation", MessageBoxButtons.OKCancel, MessageBoxIcon.Question) = System.Windows.Forms.DialogResult.OK Then
				CType(arg.Page, XtraTabPage).PageVisible = False
			End If
		End Sub
	End Class
End Namespace

File: CS/ConfirmTabClose/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraTab;
using DevExpress.XtraTab.ViewInfo;

namespace ConfirmTabClose {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void xtraTabControl1_CloseButtonClick(object sender, EventArgs e) {
            XtraTabControl tabControl = (XtraTabControl)sender;
            ClosePageButtonEventArgs arg = (ClosePageButtonEventArgs)e;
            if(MessageBox.Show(string.Format("Close {0}?", arg.Page.Text), "Confirmation", MessageBoxButtons.OKCancel, MessageBoxIcon.Question) == DialogResult.OK) {
                ((XtraTabPage)arg.Page).PageVisible = false;
            }
        }
    }
}
